File: src/main/java/com/study/platform/domain/team/model/TeamMemberRepository.java
package com.study.platform.domain.team.model;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface TeamMemberRepository {

    TeamMember save(TeamMember member);
    void delete(TeamMember member);
    List<TeamMember> findAllByTeamId(UUID teamId);
    Optional<TeamMember> findByTeamIdAndUserId(UUID teamId, UUID userId);
    boolean existsByTeamIdAndUserId(UUID teamId, UUID userId);
    long countByTeamId(UUID teamId);
    void deleteAllByTeamId(UUID teamId);
}

File: domain/src/main/kotlin/com/study/platform/domain/team/model/TeamMemberRepository.kt
package com.study.platform.domain.team.model

import java.util.UUID

interface TeamMemberRepository {

    fun save(member: TeamMember): TeamMember
    fun delete(member: TeamMember)
    fun findAllByTeamId(teamId: UUID): List<TeamMember>
    fun findAllByTeamIdIn(teamIds: Collection<UUID>): List<TeamMember>
    fun findByTeamIdAndUserId(teamId: UUID, userId: UUID): TeamMember?
    fun existsByTeamIdAndUserId(teamId: UUID, userId: UUID): Boolean
    fun countByTeamId(teamId: UUID): Long
    fun deleteAllByTeamId(teamId: UUID)
}
